assembl/static2/js/app/components/search.jsx

```diff
@@ -0,0 +1,647 @@
+  HasChildQuery,
+  Hits,
+  HitsStats,
+  InitialLoader,
+  Layout,
+  LayoutBody,
+  LayoutResults,
+  MenuFilter,
+  Pagination,
+  Panel,
+  RangeFilter,
+  RangeQuery,
+  ResetFilters,
+  SearchBox,
+  SearchkitManager,
+  SearchkitProvider,
+  // SelectedFilters,
+  SideBar,
+  TagFilterConfig,
+  TagFilter,
+  TermQuery,
+  TopBar
+} from 'searchkit';
+import get from 'lodash/get';
+import truncate from 'lodash/truncate';
+
+// Keep the style import here. The reason why it's not in main.scss is because
+// we create a searchv1 bundle that includes only the Search component and its
+// local styles for v1. There is an additional searchv1.scss file that overrides
+// some styles for v1
+
+import '../../../css/views/search.scss';
+
+import DateRangeFilter from './search/DateRangeFilter';
+import FilteredSortingSelector from './search/SortingSelector';
+import ProfileLine from './common/profileLine';
+import { getConnectedUserId, getDebateId, getLocale } from '../reducers/contextReducer';
+import { getPermissionsForConnectedUser } from '../reducers/usersReducer';
+import { canUseExpertInterface } from '../utils/permissions';
+
+const FRAGMENT_SIZE = 400;
+
+const truncateText = (text) => {
+  return truncate(text,
+    { length: FRAGMENT_SIZE, separator: ' ', omission: ' [...]' });
+};
+
+const highlightedTextOrTruncatedText = (hit, field) => {
+  let text = get(hit, `highlight.${field}`);
+  if (text) {
+    if (Array.isArray(text)) {
+      // take the first highlight fragment
+      text = text[0];
+    }
+    return text;
+  }
+
+  text = truncateText(hit._source[field]);
+  return text;
+};
+
+let Link;
+let getUrl;
+if (__resourceQuery) { // v1
+  // const querystring = require('querystring');
+  // const params = querystring.parse(__resourceQuery.slice(1));
+  // if (params.v === '1') {
+  Link = (props) => {
+    return <a href={props.to} dangerouslySetInnerHTML={props.dangerouslySetInnerHTML} />;
+  };
+  getUrl = (hit) => {
+    const slug = document.getElementById('discussion-slug').value;
+    const id = hit._source.id;
+    switch (hit._type) {
+    case 'synthesis':
+      return `${slug}/posts/local:Content/${id}`;
+    case 'user':
+      return undefined;
+    case 'idea':
+      return `${slug}/idea/local:Idea/${id}`;
+    default: // post
+      return `${slug}/posts/local:Content/${id}`;
+    }
+  };
+  // }
+} else {
+  Link = require('react-router').Link;  // eslint-disable-line
+  getUrl = (hit) => {
+    const id = hit._source.id;
+    switch (hit._type) {
+    case 'synthesis':
+      return `posts/${id}`;
+    case 'user':
+      return `profile/${id}`;
+    case 'idea':
+      return `ideas/${id}`;
+    default: // post
+      return `posts/${id}`;
+    }
+  };
+}
+
+const PublishedInfo = (props) => {
+  const { date, userId, userName, className } = props;
+  return (
+    <div className={className}>
+      <Translate value="search.published_the" />{' '}<Localize value={date} dateFormat="date.format" />
```

Comment:
"published_the" and "Published the" is not good English I guess. Maybe replace with something like "publication_date" or "published_on".